src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1043,6 +1039,10 @@ public Consumer createConsumerFromDTO(ConsumerDTO consumer, ConsumerType type, P
                     "");
             }
 
+            if (keys.size() > 0) {
+                consumerBindUtil.handleActivationKeys(consumerToCreate, keys, owner.isAutobindDisabled());
+            }
+
```

Comment:
I figured it would be a good idea to handle the service level logic first and then go into the handleActivationKeys to help prevent issues with a null service level value on the Consumer object.

Reply:
:+1: